jetty-websocket/jetty-websocket-server/src/main/java/org/eclipse/jetty/websocket/server/JettyWebSocketServletContainerInitializer.java

```diff
@@ -62,41 +70,21 @@ public void doStart()
         }
     }
 
-    public static void configure(ServletContextHandler contextHandler)
+    public static void configure(ServletContextHandler contextHandler) throws ServletException
     {
         JettyWebSocketServletContainerInitializer sci = new JettyWebSocketServletContainerInitializer();
         JettyWebSocketEmbeddedStarter starter = new JettyWebSocketEmbeddedStarter(sci, contextHandler);
         contextHandler.addBean(starter);
     }
 
     @Override
-    public void onStartup(Set<Class<?>> c, ServletContext ctx) throws ServletException
+    public void onStartup(Set<Class<?>> c, ServletContext servletContext) throws ServletException
     {
-        // TODO why doesn't the javax side share this approach?
-        List<WebSocketServletFrameHandlerFactory> factories = (List<WebSocketServletFrameHandlerFactory>)ctx
-            .getAttribute(WebSocketServletFrameHandlerFactory.ATTR_HANDLERS);
-        if (factories == null)
-        {
-            factories = new ArrayList<>();
-            ctx.setAttribute(WebSocketServletFrameHandlerFactory.ATTR_HANDLERS, factories);
-        }
-
-        Executor executor = (Executor)ctx.getAttribute("org.eclipse.jetty.server.Executor");
-        if (executor == null)
-        {
-            try
-            {
-                QueuedThreadPool threadPool = new QueuedThreadPool();
-                threadPool.setName("Jetty-WebSocketServer");
-                threadPool.start();
-                executor = threadPool;
-            }
-            catch (Exception e)
-            {
-                throw new ServletException("Unable to start internal Executor", e);
-            }
-        }
+        WebSocketMapping mapping = WebSocketMapping.ensureMapping(servletContext);
```

Comment:
Missing the embedded-jetty pre-configured mappings attribute.
Where did it go?

Reply:
Let's discuss this in #3256